changelog/17768.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:change
+auth/approle: Add maximum length of 4096 for approle role_names, as this value results in HMAC calculation
```

Comment:
I am bit confused by this phrase. Do you mean that HMAC result is of length 4096?